week-3/todo-list/script.js

```diff
@@ -1,25 +1,99 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
+function populateTodoList() {
+  list.innerHTML = "";
+
   // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
+  for (const todo of todos) {
+    const toDoListElement = document.createElement("li");
+    toDoListElement.innerText = todo.task;
+    toDoListElement.classList.add(todo.task.replaceAll(" ", "-"));
 
-populateTodoList(todos);
+    const doneBtn = document.createElement("button");
+    doneBtn.innerText = "Done";
+    doneBtn.classList.add("todo-btn");
+    doneBtn.classList.add("done-btn");
+    doneBtn.classList.add("right-gap");
+
+    const deleteBtn = document.createElement("button");
+    deleteBtn.innerText = "Delete";
+    deleteBtn.classList.add("todo-btn");
+    deleteBtn.classList.add("delete-btn");
+
+    const span = document.createElement("span");
+
+    if (todo.completed) {
+      toDoListElement.style.textDecoration = "line-through";
+      toDoListElement.style.listStyleType = "disc";
+      doneBtn.innerText = "Undone";
+    }
+
+    span.appendChild(doneBtn);
+    span.appendChild(deleteBtn);
+    toDoListElement.appendChild(span);
+    list.appendChild(toDoListElement);
```

Comment:
Awesome job here!!!! You'll learn much about DOM manipulation in the other modules, but you already follow the best practices, which is to prepare everything with JS and then add **one** thing to the DOM _if possible_. You absolutely smashed it.